pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -366,3 +312,127 @@ func removeMonitorFromQuorum(context *clusterd.Context, clusterName, name string
 	logger.Infof("removed monitor %s", name)
 	return nil
 }
+
+func (c *Cluster) resolveInvalidMonitorPlacement(desiredMonCount int) (bool, error) {
+	nodeChoice, err := c.findInvalidMonitorPlacement(desiredMonCount)
+	if err != nil {
+		return true, fmt.Errorf("failed to find invalid mon placement %+v", err)
+	}
+
+	// no violation found
+	if nodeChoice == nil {
+		return false, nil
+	}
+
+	for name, nodeInfo := range c.mapping.Node {
+		if nodeInfo.Name == nodeChoice.Node.Name {
+			logger.Infof("rebalance: rescheduling mon %s from node %s", name, nodeInfo.Name)
+			c.failMon(len(c.clusterInfo.Monitors), desiredMonCount, name)
```

Comment:
the naming of `failMon` is misleading. the selected monitor will be failed only if the monitor count is larger than the target, otherwise it will try to fail over / move the monitor to a different node.

Reply:
The behavior I've observed is that the new mon is created and added to the cluster before the current one is removed. Does that seem sufficient @leseb ?

Reply:
Yes, it is. Thanks.